app/main/views/drafts.py

```diff
@@ -139,6 +139,8 @@ def list_draft_services():
         framework = Framework.query.filter(
             Framework.slug == framework_slug
         ).first()
```

Comment:
The reason I didn't use `first_or_404` here was that the error message didn't give any indication as to what had gone wrong. Given that multiple things could be causing the 404 I think a specific error message is more useful.